lib/coney_island/performer.rb

```diff
@@ -22,13 +24,16 @@ def method_missing(method_name, *args)
     module ClassMethods
 
       def set_background_defaults(work_queue: nil, delay: nil, timeout: nil)
-        self.coney_island_settings[:work_queue] = work_queue
-        self.coney_island_settings[:delay] = delay
-        self.coney_island_settings[:timeout] = timeout
+        # This works as intended, subclasses get their own specific configurations
+        # if they pass any and inherit config from base classes but we should
+        # treat these defaults better or whatever isn't passed gets merged
```

Comment:
@darkside, would it make sense to wrap each setting in a conditional so that it only gets overwritten if it's passed in?

```ruby
if work_queue.present?
  self.coney_island_settings = get_coney_settings.merge work_queue: work_queue
end
if delay.present?
...